### PR TITLE
refactor(benchpress): convert src and test to typescript

### PR DESCRIPTION
**Pull request by jeffbcross**

### CODE REVIEW

File: modules/benchpress/src/firefox_extension/lib/main.ts [hunk 1]
```diff
@@ -48,14 +45,8 @@ mod.PageMod({
...
-    });
-    worker.port.on('stopAndRecord', function(filePath) {
-      stopAndRecord(filePath);
-    });
-    worker.port.on('forceGC', function() {
-      forceGC();
-    });
+    worker.port.on('startProfiler', function() { startProfiler(); });
+    worker.port.on('stopAndRecord', function(filePath) { stopAndRecord(filePath); });
+    worker.port.on('forceGC', function() { forceGC(); });
```
Comment (jeffbcross) [inline modules/benchpress/src/firefox_extension/lib/main.ts]: convert all to fat arrow

File: modules/benchpress/src/firefox_extension/data/installed_script.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+declare var exportFunction;
+declare var self: Window;
```
Comment (jeffbcross) [inline modules/benchpress/src/firefox_extension/data/installed_script.ts]: Not needed

File: modules/benchpress/src/common_options.ts [hunk 3]
```diff
@@ -40,10 +40,11 @@ var _NOW = new OpaqueToken('Options.now');
...
-  bind(_DEFAULT_DESCRIPTION).toValue({}),
+  bind(_DEFAULT_DESCRIPTION)
+      .toValue({}),
   bind(_SAMPLE_DESCRIPTION).toValue({}),
   bind(_FORCE_GC).toValue(false),
   bind(_PREPARE).toValue(false),
   bind(_MICRO_METRICS).toValue({}),
-  bind(_NOW).toValue( () => DateWrapper.now() )
+  bind(_NOW).toValue(() => DateWrapper.now())
 ];
```
Comment (jeffbcross) [inline modules/benchpress/src/common_options.ts]: Add newline

File: modules/benchpress/src/firefox_extension/lib/test_helper.ts [hunk 4]
```diff
@@ -19,15 +19,14 @@ exports.getFirefoxProfile = function(extensionPath) {
...
-      var multiCapabilities = [{
-        browserName: 'firefox',
-        firefox_profile : encodedProfile
-      }];
-      deferred.resolve(multiCapabilities);
-    });
-  });
+  firefoxProfile.addExtensions(
+      [extensionPath],
+      function() {
```
Comment (jeffbcross) [inline modules/benchpress/src/firefox_extension/lib/test_helper.ts]: Make fat arrow

File: modules/benchpress/src/metric/perflog_metric.ts [hunk 5]
```diff
@@ -81,47 +93,43 @@ export class PerflogMetric extends Metric {
...
-    return this._endMeasure(true).then( (measurValues) => {
-        return this._driverExtension.gc()
-          .then( (_) => this._endMeasure(restartMeasure) )
-          .then( (forceGcMeasureValues) => {
+  _endPlainMeasureAndMeasureForceGc(restartMeasure: boolean) {
+    return this._endMeasure(true).then((measurValues) => {
+      return this._driverExtension.gc()
+          .then((_) => this._endMeasure(restartMeasure))
+          .then((forceGcMeasureValues) => {
             StringMapWrapper.set(measurValues, 'forcedGcTime', forceGcMeasureValues['gcTime']);
```
Comment (jeffbcross) [inline modules/benchpress/src/metric/perflog_metric.ts]: Fix spelling of "measurValues"

**Comment by tbosch** [comment]
Let's pair on this tomorrow to land it.

File: modules/benchpress/src/reporter/console_reporter.ts [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+var _BINDINGS = [
+  bind(ConsoleReporter)
+      .toFactory((columnWidth, sampleDescription, print) =>
+                     new ConsoleReporter(columnWidth, sampleDescription, print),
+                 [_COLUMN_WIDTH, SampleDescription, _PRINT]),
+  bind(_COLUMN_WIDTH).toValue(18),
+  bind(_PRINT).toValue(print)
+];
+var _PRINT = new OpaqueToken('ConsoleReporter.print');
+var _COLUMN_WIDTH = new OpaqueToken('ConsoleReporter.columnWidht');
```
Comment (jeffbcross) [inline modules/benchpress/src/reporter/console_reporter.ts]: Fix spelling

**Comment by jeffbcross** [comment]
fixes pushed. please be happy, Travis.

**Pull request merged**